src/containerapp/azext_containerapp/_clients.py

```diff
@@ -1513,6 +1513,23 @@ def extract_path_from_filename(path, filename):
             return path.rstrip("/") + "/" + path_in_filename.lstrip('/'), filename
 
 
+class SessionCustomContainerPreviewClient():
+    # pylint: disable=too-few-public-methods
+    session_dp_api_version = "2025-02-02-preview" # may be different from ACA CP
+
+    @classmethod
+    def stop_session(cls, cmd, identifier, session_pool_endpoint):
+        url_fmt = "{}/.management/stopSession?identifier={}&api-version={}"
+        request_url = url_fmt.format(
+            session_pool_endpoint,
+            identifier,
+            cls.session_dp_api_version)
+
+        r = send_raw_request(cmd.cli_ctx, "POST", request_url, resource=SESSION_RESOURCE)
```

Comment:
can we get the pool from cp first and check whether it is a custom container pool, and fail on client side when pool type is not custom container. Otherwise I suppose it would return 404 for python pool, which is not good experience.